config/codeartifact/config.go

```diff
@@ -0,0 +1,47 @@
+// SPDX-FileCopyrightText: 2024 The Crossplane Authors <https://crossplane.io>
+//
+// SPDX-License-Identifier: CC0-1.0
+
+package codeartifact
+
+import (
+	"github.com/crossplane/upjet/pkg/config"
+
+	"github.com/upbound/provider-aws/config/common"
+)
+
+// Configure adds configurations for the codeartifact group.
+func Configure(p *config.Provider) {
+	p.AddResourceConfigurator("aws_codeartifact_domain", func(r *config.Resource) {
+		r.References["encryption_key"] = config.Reference{
+			TerraformName: "aws_kms_key",
+			Extractor:     common.PathARNExtractor,
+		}
+	})
+
+	p.AddResourceConfigurator("aws_codeartifact_domain_permissions_policy", func(r *config.Resource) {
+		r.References["domain"] = config.Reference{
+			TerraformName: "aws_codeartifact_domain",
+			Extractor:     "ExtractDomainName()",
+		}
+	})
+
+	p.AddResourceConfigurator("aws_codeartifact_repository", func(r *config.Resource) {
+		r.References["domain"] = config.Reference{
+			TerraformName: "aws_codeartifact_domain",
+			Extractor:     "ExtractDomainName()",
+		}
+	})
+
+	p.AddResourceConfigurator("aws_codeartifact_repository_permissions_policy", func(r *config.Resource) {
+		r.References["domain"] = config.Reference{
+			TerraformName: "aws_codeartifact_domain",
+			Extractor:     "ExtractDomainName()",
```

Comment:
```suggestion
			Extractor:     `github.com/crossplane/upjet/pkg/resource.ExtractParamPath("domain",true)`,
```

Reply:
Updated in https://github.com/crossplane-contrib/provider-upjet-aws/pull/1368/commits/2f037e2a74d6ccbc4a0d79fb4239b8ffc682a9e5